halotools/mock_observables/pair_counters/marked_npairs_3d.py

```diff
@@ -195,7 +197,7 @@ def _marked_npairs_process_weights(sample1, sample2, weights1, weights2, weight_
     _converted_to_2d_from_1d = False
     # First convert weights1 into a 2-d ndarray
     if weights1 is None:
-        weights1 = np.ones((npts_sample1, 1), dtype=np.float64)
+        weights1 = np.ones(correct_shape1, dtype=np.float64)
```

Comment:
Previously, if the weight function expected multiple weights, but you didn't pass the weights, this would break. I'm pretty sure that this is what you want. The same change is made in a couple of other places

Reply:
Nice catch!